### PR TITLE
chore(deps): bump alloy

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

New version of alloy has been released.

## Solution

Bump alloy to 0.3

### CODE REVIEW

**Review by onbjerg — APPROVED** [review]

File: crates/common/fmt/src/ui.rs [hunk 1]
```diff
@@ -338,7 +338,7 @@ to                   {}
 transactionIndex     {}
 type                 {}
 value                {}
-yParity              {}{}",
+yParity              {}",
```
Comment (klkvr) [inline crates/common/fmt/src/ui.rs:341]: We should still format additional fields, I think we can just impl `UIfmt` for `WithOtherFields<Transaction>`?

**Comment by mattsse** [comment]
what's missing here?

**Comment by yash-atreya** [comment]
> There's still some work to do
> 
> 
> 
> Merged in https://github.com/foundry-rs/foundry/pull/8627 which includes updated for alloy-core
> 
> 
> 
> Leaving it up to you @yash-atreya 

Yeah

File: crates/evm/core/src/opts.rs [hunk 2]
```diff
@@ -75,18 +76,19 @@ impl EvmOpts {
...
+            Ok(self.fork_evm_env::<alloy_provider::network::AnyNetwork>(fork_url).await?.0)
         } else {
             Ok(self.local_evm_env())
         }
     }
 
     /// Returns the `revm::Env` that is configured with settings retrieved from the endpoint.
     /// And the block that was used to configure the environment.
-    pub async fn fork_evm_env(
+    pub async fn fork_evm_env<N: Network>(
```
Comment (DaniPopes) [inline crates/evm/core/src/opts.rs:87]: N is not used

File: crates/evm/core/src/fork/init.rs [hunk 3]
```diff
@@ -29,7 +32,7 @@ pub async fn environment<N: Network, T: Transport + Clone, P: Provider<T, N>>(
         provider.get_chain_id(),
         provider.get_block_by_number(BlockNumberOrTag::Number(block_number), false)
     )?;
-    let block = if let Some(block) = block {
+    let block: <N as Network>::BlockResponse = if let Some(block) = block {
```
Comment (DaniPopes) [inline crates/evm/core/src/fork/init.rs:35]: type hint unnecessary

**Comment by mattsse** [comment]
can we allow evmole in deny.toml for now, we shouldn't block this pr

File: crates/anvil/Cargo.toml [hunk 4]
```diff
@@ -122,6 +122,7 @@ alloy-pubsub.workspace = true
 foundry-test-utils.workspace = true
 similar-asserts.workspace = true
 tokio = { workspace = true, features = ["full"] }
+op-alloy-rpc-types = "0.2.8"
```
Comment (mattsse) [inline crates/anvil/Cargo.toml:125]: please move to workspace, all of alloy deps should be workspace levle

File: crates/anvil/core/src/eth/transaction/optimism.rs
Comment (mattsse) [inline crates/anvil/core/src/eth/transaction/optimism.rs:1]: ah followup task would be to replace this with the alloy type

**Review by mattsse — CHANGES_REQUESTED** [review]
nit

ptal @klkvr

**Comment by cdump** [comment]
I've released evmole 0.5.0 with updated alloy. A small patch was required for foundry-rs - https://github.com/cdump/evmole/issues/13#issuecomment-2331832673

I haven't created a PR here because of a lot of Cargo.lock changes related to the alloy upgrade, so it's better to apply it in this existing PR

File: crates/anvil/src/eth/otterscan/api.rs [hunk 5]
```diff
@@ -176,7 +180,19 @@ impl EthApi {
...
+                let inner_txs =
+                    transactions.into_transactions().map(|t| t.inner).collect::<Vec<Transaction>>();
+
+                let mut inner_block =
+                    Block { header, uncles, size, withdrawals, ..Default::default() };
+
+                inner_block.transactions = BlockTransactions::from(inner_txs);
+
+                self.build_ots_block_tx(inner_block, page, page_size).await
+            }
```
Comment (klkvr) [inline crates/anvil/src/eth/otterscan/api.rs:195]: ots types have tx and block generics too, we should update `build_ots_block_tx` instead

File: crates/common/fmt/src/ui.rs [hunk 6]
```diff
@@ -1058,7 +1056,10 @@ value                0".to_string();
           }
         );
 
-        let block: Block = serde_json::from_value(json).unwrap();
+        let block: WithOtherFields<Block<WithOtherFields<Transaction>>> =
+            serde_json::from_value(json).unwrap();
+
+        // let block: WithOtherFields<Block> = WithOtherFields::new(block);
```
Comment (klkvr) [inline crates/common/fmt/src/ui.rs:1062]:
```suggestion
```

**Review by klkvr — CHANGES_REQUESTED** [review]

**Comment by DaniPopes** [comment]
Released alloy-core 0.8.1 https://github.com/alloy-rs/core/releases/tag/v0.8.1

**Review by DaniPopes — APPROVED** [review]

File: crates/anvil/src/eth/api.rs [hunk 7]
```diff
@@ -88,7 +88,6 @@ use std::{
 
 /// The client version: `anvil/v{major}.{minor}.{patch}`
 pub const CLIENT_VERSION: &str = concat!("anvil/v", env!("CARGO_PKG_VERSION"));
```
Comment (DaniPopes) [inline crates/anvil/src/eth/api.rs:90]:
```suggestion
pub const CLIENT_VERSION: &str = concat!("anvil/v", env!("CARGO_PKG_VERSION"));

```

**Review by mattsse — APPROVED** [review]

**Review by klkvr — APPROVED** [review]

**Pull request merged**